src/test/java/uk/gov/laa/gpfd/service/UserServiceTest.java

```diff
@@ -2,68 +2,177 @@
-
-    when(mockAzureGraphClient.getGraphUserDetails(mockOAuth2Client)).thenReturn(graphUser);
-
-    UserDetails result = userService.getUserDetails(mockOAuth2Client);
-
-    assertEquals(graphUser.userPrincipalName, result.userPrincipalName());
-    assertEquals(graphUser.givenName, result.givenName());
-    assertEquals(graphUser.surname, result.surname());
-    assertEquals(graphUser.preferredName, result.preferredName());
-    assertEquals(graphUser.mail, result.email());
-  }
-
-  @Test
-  void whenNullGraphUserIsReturnedUserServiceExceptionIsThrown() {
-    assertThrows(AuthUserNotFoundException.class, () -> userService.getUserDetails(mockOAuth2Client));
-  }
-
-  @Test
-  void whenGraphExceptionThenUserServiceExceptionIsThrown() {
-    when(mockAzureGraphClient.getGraphUserDetails(mockOAuth2Client)).thenThrow(new ClientException("client error", null));
-
-    assertThrows(UserServiceException.class, () -> userService.getUserDetails(mockOAuth2Client));
-  }
+    @Mock
+    OAuth2AuthorizedClient mockOAuth2Client;
+
+    @Mock
+    AzureGraphClient mockAzureGraphClient;
+
+    @InjectMocks
+    UserService userService;
+
+    @Test
+    void shouldReturnValidUserWhenGraphReturnsCompleteUserDetails() {
+        var graphUser = AzureGraphUserTestDataFactory.aValidUserSmallUser();
+        when(mockAzureGraphClient.getGraphUserDetails(mockOAuth2Client)).thenReturn(graphUser);
+
+        var result = userService.getUserDetails(mockOAuth2Client);
+
+        assertEquals(graphUser.userPrincipalName, result.userPrincipalName());
+        assertEquals(graphUser.givenName, result.givenName());
+        assertEquals(graphUser.surname, result.surname());
+        assertEquals(graphUser.preferredName, result.preferredName());
+        assertEquals(graphUser.mail, result.email());
+    }
+
+    @Test
+    void shouldThrowAuthUserNotFoundExceptionWhenGraphReturnsNullUser() {
+        assertThrows(AuthUserNotFoundException.class, () -> userService.getUserDetails(mockOAuth2Client));
+    }
+
+    @Test
+    void shouldThrowUserServiceExceptionWhenClientExceptionOccurs() {
+        when(mockAzureGraphClient.getGraphUserDetails(mockOAuth2Client)).thenThrow(new ClientException("client error", null));
+
+        // Then
+        assertThrows(UserServiceException.class, () -> userService.getUserDetails(mockOAuth2Client));
+    }
+
+    @Test
+    void shouldReturnErrorMessageWhenNoDetailsAreReturnedFromGraph() {
+        // Given
+        when(mockAzureGraphClient.getGraphUserDetails(mockOAuth2Client)).thenReturn(null);
+
+        // When
+        var exception = assertThrows(AuthUserNotFoundException.class, () -> userService.getUserDetails(mockOAuth2Client));
+
+        // Then
+        assertEquals("No user details returned from Graph", exception.getMessage());
+    }
+
+    @Test
+    void shouldThrowAuthUserNotFoundExceptionWhenGraphUserDataIsIncomplete() {
+        // Given
+        var incompleteUser = new User(); // Missing userPrincipalName and email
```

Comment:
Currently reads as if email is also mandatory - doesn't look to be the case. Can test be updated to use a User that is only missing mandatory fields (userPrincipalName)